wp-job-manager-functions.php

```diff
@@ -1703,3 +1708,40 @@ function job_manager_get_salary_unit_options( $include_empty = true ) {
 	 */
 	return apply_filters( 'job_manager_get_salary_unit_options', $options, $include_empty );
 }
+
+/**
+ * Checks if the job expiry can be extended.
+ * This is true if the job is public, the option to extend is set and the job expires within 5 days.
+ *
+ * @since $$next_version$$
+ *
+ * @param int|WP_Post $job The job or job ID.
+ *
+ * @return bool
+ */
+function job_manager_job_can_be_renewed( $job ) {
+	$job    = get_post( $job );
+	$status = get_post_status( $job );
+
+	/*
+	 * `WP_Job_Manager_Post_Types::instance()->get_job_expiration( $job )->getTimestamp()` seems off by a day.
+	 * The following expression is equivalent to: `strtotime( get_post_meta( $job->ID, '_job_expires', true ) )`
+	 */
+	$expiry = WP_Job_Manager_Post_Types::instance()->get_job_expiration( $job )->getTimestamp() - DAY_IN_SECONDS;
+
+	// If there is no expiry, then relisting is not necessary.
+	if ( ! $expiry ) {
```

Comment:
`$expiry` can be `DateTimeImmutable` or `false`. Although it works, I think we should avoid implicit type conversion as they can cause hard-to-track bugs.

Reply:
https://github.com/Automattic/WP-Job-Manager/pull/2422/commits/b42c439429cefa978335297df2635830c92b30f3